jarvis/jasper_v0.8.2_chkpt_to_onnx.py

```diff
@@ -0,0 +1,74 @@
+import nemo
+import nemo_asr
+from nemo_asr.parts.jasper import MaskedConv1d
+from ruamel.yaml import YAML
+import onnx
+import torch
+
+
+def get_parser():
+    parser = argparse.ArgumentParser(
+            description="Convert 0.8.2 jasper checkpoint to onnx")
+    parser.add_argument(
+            "--config", default=None, type=str, required=True,
+            help="Config from nemo")
+    parser.add_argument(
+            "--nn_encoder", default=None, type=str, required=True,
+            help="Path to the nn encoder checkpoint.")
+    parser.add_argument(
+            "--nn_decoder", default=None, type=str, required=True,
+            help="Path to the nn encoder checkpoint.")
+    parser.add_argument(
+            "--onnx_encoder", default=None, type=str, required=True,
+            help="Path to the onnx encoder output.")
+    parser.add_argument(
+            "--nn_decoder", default=None, type=str, required=True,
+            help="Path to the onnx decoder output.")
+    return parser
+
+
+def main(config_file, nn_encoder, nn_decoder, nn_onnx_encoder, nn_onnx_decoder):
+    yaml = YAML(typ="safe")
+    with open(config_file) as f:
+        jasper_model_definition = yaml.load(f)
+
+    jasper_encoder = nemo_asr.JasperEncoder(
+        feat_in=64,
+        **jasper_model_definition['JasperEncoder'])
+    jasper_decoder = nemo_asr.JasperDecoderForCTC(
+        feat_in=1024, num_classes=len(jasper_model_definition['labels']))
+
+    # hack to map the state dict of previous version
+    ckpt = torch.load(nn_encoder)
```

Comment:
Let's remove conversion and assume compatible checkpoints (which we will publish). 
Also, make it as small and concise as possible. 
```python
import nemo
import nemo_asr
from ruamel.yaml import YAML
import onnx
import torch
yaml = YAML(typ="safe")
with open("jasper10x5dr.yaml") as f:
    jasper_model_definition = yaml.load(f)

nf = nemo.core.NeuralModuleFactory()

jasper_encoder = nemo_asr.JasperEncoder(
    feat_in=64,
    **jasper_model_definition['JasperEncoder'])
jasper_decoder = nemo_asr.JasperDecoderForCTC(
    feat_in=1024, num_classes=len(jasper_model_definition['labels']))
jasper_encoder.restore_from("Jasper10x5DR_Encoder.pt")
jasper_decoder.restore_from("Jasper10x5DR_Decoder.pt")

nf.deployment_export(jasper_encoder, "./nn_encoder.onnx",
                     nemo.core.neural_factory.DeploymentFormat.ONNX,
                     (torch.zeros(1, 64, 256, dtype=torch.float, device="cuda:0"), torch.zeros(1, dtype=torch.int, device="cuda:0")))
nf.deployment_export(jasper_decoder, "./nn_decoder.onnx",
                     nemo.core.neural_factory.DeploymentFormat.ONNX,
                     (torch.zeros(1, 1024, 128, dtype=torch.float, device="cuda:0")))

```